### PR TITLE
Release 2.1.0

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Release 2.0.2
| Type?         | improvement
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | N/A
| How to test?  | Release test in correlation with https://github.com/PrestaShop/PrestaShop/pull/22032.

### CODE REVIEW

**Review by PierreRambaud — DISMISSED** [review]

**Comment by sarahdib** [comment]
@Progi1984 during the upgrade process this should be done https://github.com/PrestaShop/ps_searchbar/pull/12#issuecomment-570504717 in order to have the icon 

Capture d’écran 2020-11-18 à 17 03 37
thumbnail should appear in the research.

**Review by jolelievre — CHANGES_REQUESTED** [review]
I block this one for safety to avoid merging/releasing it too sone

It seems the modification for the new search bar are not compatible with the current classic theme
We need to remove files from the classic theme so it will only be done in 1.7.8 (unless we make a quick fix for 1.7.7)

In any case the module can only work properly with a recent/future version of PrestaShop so we need to update its compatibility range so it's not a patch update, nor even a minor one but a **major** one

tldr: The version should be 3.0.0

**Comment by PierreRambaud** [comment]
> I block this one for safety to avoid merging/releasing it too sone
> 
> It seems the modification for the new search bar are not compatible with the current classic theme
> We need to remove files from the classic theme so it will only be done in 1.7.8 (unless we make a quick fix for 1.7.7)
> 
> In any case the module can only work properly with a recent/future version of PrestaShop so we need to update its compatibility range so it's not a patch update, nor even a minor one but a **major** one
> 
> tldr: The version should be 3.0.0

Regarding semver, changing a dependency version range is not part of the public API, otherwise, each time we upgrade dependencies, we must do a major version :thinking: 
https://semver.org/#what-should-i-do-if-i-update-my-own-dependencies-without-changing-the-public-api

The PrestaShop software is able to decline an installation of incompatible module.

**Comment by Progi1984** [comment]
@sarahdib We have found the problem. The override on the classic theme has not been removed. So you can test with the PR : https://github.com/PrestaShop/PrestaShop/pull/22032

**Comment by sarahdib** [comment]
It's OK with the PR #22032 on core using develop (1.7.8.)version

**Comment by Progi1984** [comment]
@atomiix Your PR #12 is only available on 178. Could you fix the `ps_versions_compliancy` ?

**Comment by atomiix** [comment]
@Progi1984 you mean doing a new PR to update `ps_versions_compliancy`?

**Comment by Progi1984** [comment]
Actually this PR is not compatible with 177, only with develop with PrestaShop/PrestaShop#22032.

**Comment by matks** [comment]
@Progi1984 @atomiix should we bump milestone and release number to 2.1.0 or 3.00 instead of 2.0.2 ? it's quite weird for module user to see a patch version bumping main core requirement

**Comment by Progi1984** [comment]
@matks The API is not broken so no way.

**Comment by matks** [comment]
> @matks The API is not broken so no way.

I disagree with SemVer "no change in API, no bump" policy and many people think the same (see https://github.com/semver/semver/issues/148)

Doctrine is famous for writing ["Why dropping PHP support in a minor version is not a BC break"](https://www.doctrine-project.org/2017/07/25/php-7.1-requirement-and-composer.html) but this has proven to be quite painful for its users over the years 😅  I suggest we do not follow the same path.

**Comment by matks** [comment]
@Progi1984 Anyways I just noticed https://github.com/PrestaShop/ps_searchbar/pull/12 is milestoned for next release. https://github.com/PrestaShop/ps_searchbar/pull/12 is not a bugfix, it is a feature. This means that next release must be at least 2.1.0, not 2.0.2 .

**Comment by Progi1984** [comment]
@matks Done

**Review by jolelievre — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]
